src/projects/projects.service.ts

```diff
@@ -14,33 +14,84 @@ export class ProjectsService {
     return exclude(project, ['id', 'userId']);
   }
 
-  async create(user: User, project: CreateProjectDto): Promise<SanitizedProject> {
-    return this.sanitizeOutput(await this.prisma.project.create({
+  async create(userId: number, project: CreateProjectDto): Promise<SanitizedProject> {
+    const createdProject = await this.prisma.project.create({
       data: {
         name: project.name,
         uuid: uuidv4(),
-        userId: user.id,
-      }
-    }));
+        userId: userId,
+      },
+    });
+
+    const sanitizedOutput = this.sanitizeOutput(createdProject);
+
+    return {
+      ...sanitizedOutput,
+      uuid: createdProject.uuid,
+    };
   }
 
-  async findOne(uuid: string): Promise<SanitizedProject|null> {
-    const project = await this.prisma.project.findUnique({
-      // FIXME: ensure one can only find their projects and not somebody else's
-      where: {uuid},
+  async findOne(uuid: string, userId: number): Promise<SanitizedProject | null> {
```

Comment:
Could you throw an error if the project is not found instead of returning `null`?

Same in other methods

Resource: [findFirstOrThrow](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findfirstorthrow)

Reply:
nice ! i replaced all the logic for findFirst to findFirstOrThrow

Reply:
don't forget what you return from the function